azurerm/resource_arm_sql_extended_server_blob_auditing_policies.go

```diff
@@ -0,0 +1,218 @@
+				return fmt.Errorf("Error checking for presence of existing blob auditing policies of SQL Extended Server %q  Blob Auditing Policies(Resource Group %q): %+v", serverName, resGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_sql_extended_server_blob_auditing_policies", *existing.ID)
+		}
+	}
+
+	var state sql.BlobAuditingPolicyState
+	switch d.Get("state").(string) {
+	case "Enabled":
+		state = sql.BlobAuditingPolicyStateEnabled
+	case "Disabled":
+		state = sql.BlobAuditingPolicyStateDisabled
+	}
+	storageEndpoint := d.Get("storage_endpoint").(string)
+	storageAccountAccessKey := d.Get("storage_account_access_key").(string)
+
+	ExtendedServerBlobAuditingPolicyProperties := sql.ExtendedServerBlobAuditingPolicyProperties{
+		State:                   state,
+		StorageEndpoint:         &storageEndpoint,
+		StorageAccountAccessKey: &storageAccountAccessKey,
+	}
+	//retention_days
+	if retentionDays, ok := d.GetOk("retention_days"); ok {
+		retentionDays := int32(retentionDays.(int))
+		ExtendedServerBlobAuditingPolicyProperties.RetentionDays = &retentionDays
+	}
+	//audit_actions_and_groups
+	if auditActionsAndGroups, ok := d.GetOk("audit_actions_and_groups"); ok {
+		auditActionsAndGroups := strings.Split(auditActionsAndGroups.(string), ",")
+		ExtendedServerBlobAuditingPolicyProperties.AuditActionsAndGroups = &auditActionsAndGroups
+	}
+	//storage_account_subscription_id
+	if storageAccountSubscriptionID, ok := d.GetOk("storage_account_subscription_id"); ok {
+		storageAccountSubscriptionID, err := uuid.FromString(storageAccountSubscriptionID.(string))
+		if err != nil {
+			return fmt.Errorf("Error transforming storage account subscription id from string to uuid:%s", err)
+		}
+		ExtendedServerBlobAuditingPolicyProperties.StorageAccountSubscriptionID = &storageAccountSubscriptionID
+	}
+	//is_storage_secondary_key_in_use
+	if isStorageSecondaryKeyInUse, ok := d.GetOk("is_storage_secondary_key_in_use"); ok {
+		isStorageSecondaryKeyInUse := isStorageSecondaryKeyInUse.(bool)
+		ExtendedServerBlobAuditingPolicyProperties.IsStorageSecondaryKeyInUse = &isStorageSecondaryKeyInUse
+	}
+	//is_azure_monitor_target_enabled
+	if isAzureMonitorTargetEnabled, ok := d.GetOk("is_azure_monitor_target_enabled"); ok {
+		isAzureMonitorTargetEnabled := isAzureMonitorTargetEnabled.(bool)
+		ExtendedServerBlobAuditingPolicyProperties.IsAzureMonitorTargetEnabled = &isAzureMonitorTargetEnabled
+	}
+	//predicate_expression
+	if predictExpression, ok := d.GetOk("predicate_expression"); ok {
+		predictExpression := predictExpression.(string)
+		ExtendedServerBlobAuditingPolicyProperties.PredicateExpression = &predictExpression
+	}
+
+	parameters := sql.ExtendedServerBlobAuditingPolicy{
+		ExtendedServerBlobAuditingPolicyProperties: &ExtendedServerBlobAuditingPolicyProperties,
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resGroup, serverName, parameters); err != nil {
```

Comment:
This method is an async method which returns a `future` object, please do not discard this object and use it for retrieving the result of the async operation.